src/components/opt-in-out/index.tsx

```diff
@@ -1,71 +1,48 @@
 import '@reach/dialog/styles.css'
-import Button from 'components/button'
-import OptOutForm from './components/opt-out-form'
-import Dialog from 'components/dialog'
-import { getLearnRedirectPath } from './helpers/get-learn-redirect-path'
-import {
-  PlatformOptionRedirectData,
-  OptInOutProps,
-  isOptInPlatformOption,
-} from './types'
 import { safeAnalyticsTrack } from 'lib/analytics'
-import makeBetaWelcomeToast from './helpers/make-beta-welcome-toast'
-import { getIoRedirectPath } from './helpers/get-io-redirect-path'
+import { OptInOutProps, isOptInPlatformOption } from './types'
+import { PLATFORM_OPTIONS, postFormData, makeBetaWelcomeToast } from './helpers'
+import { OptOutFormState } from './components/opt-out-form/types'
+import { OptOutButtonAndDialog } from './components/opt-out-button-and-dialog'
 
-// Could these go in the config? or I could source the base urls elsewhere
-export const PLATFORM_OPTIONS: PlatformOptionRedirectData = {
-  learn: {
-    base_url: 'https://learn-git-ksspike-opt-in-redirects-hashicorp.vercel.app', // FOR TESTING PURPOSES NEED TO UPDATE for - 'https://learn.hashicorp.com/'
-    getRedirectPath: getLearnRedirectPath,
-    cookieKey: 'learn-beta-opt-in',
-    cookieAnalyticsKey: 'learn-beta-opt-in-tracked',
-  },
-  'waypoint-io': {
-    base_url: 'https://www.waypointproject.io/',
-    getRedirectPath(path) {
-      return this.base_url + getIoRedirectPath(path)
-    },
-    cookieKey: 'waypoint-io-beta-opt-in',
-    cookieAnalyticsKey: 'waypoint-io-beta-opt-in-tracked',
-  },
-  'vault-io': {
-    base_url: 'https://www.vaultproject.io/',
-    getRedirectPath(path) {
-      return this.base_url + getIoRedirectPath(path)
-    },
-    cookieKey: 'vault-io-beta-opt-in',
-    cookieAnalyticsKey: 'vault-io-beta-opt-in-tracked',
-  },
-}
-
-export default function OptInOut({ platform, redirectPath }: OptInOutProps) {
+export default function OptInOutController({
+  platform,
+  redirectPath,
+}: OptInOutProps) {
   // fire toast, render button, etc
   const router = useRouter()
-  const [showDialog, setShowDialog] = useState(false)
   const optedIn = Cookies.get(PLATFORM_OPTIONS[platform].cookieKey)
-
-  const openDialog = () => setShowDialog(true)
-  const closeDialog = () => setShowDialog(false)
   const url =
     redirectPath || PLATFORM_OPTIONS[platform].getRedirectPath(router.asPath)
 
   /**
    * Handle opt out, which is passed to our opt out form,
    * and is also used in our welcome toast.
    */
-  const handleOptOut = useCallback(() => {
-    // @TODO - handle form submit
-    Cookies.remove(PLATFORM_OPTIONS[platform].cookieKey)
-    Cookies.remove(PLATFORM_OPTIONS[platform].cookieAnalyticsKey)
-    safeAnalyticsTrack('Beta Opted Out', {
-      bucket: platform,
-    })
-    window.location.assign(url)
-  }, [platform, url])
+  const handleOptOut = useCallback(
+    async (state: OptOutFormState) => {
+      await postFormData({
+        segment_anonymous_id: uuid(),
```

Comment:
Is this what's expected for this id? Just a random uuid? Not sure I have all the context here.

Reply:
I _think_ we need to get this from segment, if available:
![Screen Shot 2022-05-27 at 2 05 48 PM](https://user-images.githubusercontent.com/1289701/170774794-05d19a6a-a868-4e0d-b7a5-78ba5c76295d.png)
 
@BrandonRomano might be able to confirm.

Reply:
@BRKalow is correct here -- `window.analytics.user().anonymousId()`!

This may not be accessible, if the user has not opted-in with our consent manager.  If that's the case, don't send anything.

Reply:
@BrandonRomano what should this be?